fix: avoid hardcoded exec path

DonaldKellett opened the pull request:
Replace absolute path to executable /usr/bin/x-terminal-emulator in src/dde-file-manager-lib/shutil/fileutils.cpp with just x-terminal-emulator

Related: linuxdeepin/developer-center#3374

Hopefully I got the commit message format correct this time. If so, please review this pull request and approve if there are no outstanding issues, thanks :smile:

DonaldKellett committed:
fix: avoid hardcoded exec path

Replace absolute path to executable /usr/bin/x-terminal-emulator in src/dde-file-manager-lib/shutil/fileutils.cpp with just x-terminal-emulator

itsXuSt commented on src/dde-file-manager-lib/shutil/fileutils.cpp:
```diff
@@ -1220,11 +1220,11 @@ bool FileUtils::isFileManagerSelf(const QString &desktopFile)
-    } else if (QFileInfo::exists(debian_x_term_emu)) {
+    } else if (!debian_x_term_emu.isEmpty()) {
         return debian_x_term_emu;
     }
```
add `else` block to return the default xterm path makes code more elegant

DonaldKellett replied:
Thanks for the suggestion, implemented :smile:

DonaldKellett committed:
refactor: move return statement into else branch

Add else block to return the default xterm path in src/dde-file-manager-lib/shutil/fileutils.cpp

max-lvs reviewed (approved):
thanks，ok for me
